### PR TITLE
Docx to pdf

### PR DESCRIPTION
**Pull request by mthembbu**

### CODE REVIEW

File: api/app/invitationletter/generator.py [hunk 1]
```diff
@@ -1,9 +1,91 @@
...
+from app.utils import pdfconvertor
+from app.events.models import Event
+from app import db
+
+OFFER_EMAIL_BODY = """
+Dear {user_title} {first_name} {last_name},
+
+Congratulations! You've been selected to attend the {event_name}!
+
+Please see the attached document below for your acceptance of offer: {host}/offer
```
Comment (jaderabbit) [inline api/app/invitationletter/generator.py]: @mthembbu I remember us discussing what this email copy should be. WOuld you update?

File: api/app/invitationletter/generator.py [hunk 2]
```diff
@@ -1,9 +1,91 @@
...
+    )
+
+    document.write(invitation_letter)
+
+    # Todo: converting a generated letter into a pdf
+    pdfconvertor.convert_to(folder='app/invitationletter/letter', source=invitation_letter)
 
-    # TODO #328
+    event = db.session.query(Event).get(event_id)
+    if not event:
```
Comment (jaderabbit) [inline api/app/invitationletter/generator.py]: If no event is set, this should not run - you should throw an error instead.

File: api/app/invitationletter/generator.py [hunk 3]
```diff
@@ -1,9 +1,91 @@
...
+        subject = 'See Attachment'
+    else:
+        subject = "Invitation to " + event.name
```
Comment (jaderabbit) [inline api/app/invitationletter/generator.py]: Invitation Letter For event.name

File: api/app/utils/pdfconvertor.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+    LOGGER.debug('successfully converted to pdf...')
+    if filename is None:
+        raise LibreOfficeError(process.stdout.decode())
+    else:
+        return filename.group(1)
+
+
+def libreoffice_exec():
+    # TODO: Provide support for more platforms
```
Comment (jaderabbit) [inline api/app/utils/pdfconvertor.py]: we only need to support linux :)

File: api/app/invitationletter/generator.py [hunk 5]
```diff
@@ -1,9 +1,91 @@
...
+        residential_address=residential_address,
+        passport_name=passport_name,
+        passport_no=passport_no,
+        passport_issued_by=passport_issued_by,
+        to_date=to_date,
+        from_date=from_date,
+        country_of_residence=country_of_residence,
+        nationality=nationality,
+        date_of_birth=date_of_birth,
+        invitation_letter_sent_at=invitation_letter_sent_at
```
Comment (jaderabbit) [inline api/app/invitationletter/generator.py]: `invitation_letter_sent_at` : this shouldn't be in the merge

File: api/app/invitationletter/generator.py [hunk 6]
```diff
@@ -1,9 +1,91 @@
...
+    file_name = get_template(template_path)
+    invitation_letter = 'app/invitationletter/letter/' + file_name
+    template = 'app/invitationletter/template/' + file_name
+
+    document = MailMerge(template)
+    print(document.get_merge_fields())
+    LOGGER.debug("merge able fields.... {} .".format(document.get_merge_fields()))
+
+    document.merge(
+        firstname=firstname,
```
Comment (jaderabbit) [inline api/app/invitationletter/generator.py]: what must the fields in the merge look like?

File: api/app/invitationletter/generator.py (hunk 1 above)
Comment (mthembbu) [reply api/app/invitationletter/generator.py]: Will do.

**Comment by mthembbu** [comment]
#357

**Pull request closed without merging**